### PR TITLE
[Checkin app] Add an endpoint to lookup registration by ticket uuid

### PR DESCRIPTION
**Pull request by tomasr8**
One feedback we have gotten on the app is that the QR code is rather large and consequently sometimes difficult to scan.
We currently include lots of data in the QR code that is not strictly needed. Currently, we have:

```python
    qr_data = {
        'registrant_id': person['id'],  # accompanying person uuid if accompanying, otherwise same as registration_id
        'registration_id': person['registration'].id,
        'regform_id': registration.registration_form_id,
        'checkin_secret': checkin_secret,
        'event_id': str(registration.event.id),
        'server_url': config.BASE_URL,
        'version': 1
    }
```
(there's also some extra data when used with the site access plugin)
Since every registration has a unique ticket uuid (checkin secret), we don't need to provide `registration_id`, `regform_id` nor `event_id`.

Since we are getting rid of this data, we need a new endpoint in Indico which the app can use to lookup a registration by the ticket uuid only.

### CODE REVIEW

File: indico/modules/events/registration/controllers/api/checkin.py [hunk 1]
```diff
@@ -107,3 +107,18 @@ def _process_PATCH(self, checked_in=None, paid=None):
         if paid is not None and paid != self.registration.is_paid and self.registration.price:
             toggle_registration_payment(self.registration, paid=paid)
         return CheckinRegistrationSchema().jsonify(self.registration)
+
+
+class RHCheckinAPIRegistrationUUID(RHCheckinAPIBase):
```
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/api/checkin.py:112]: Please cover this one in the `test_unauthorized` test. I'm pretty sure we get proper access checks via inheritance, but still better to have it in the tests.

**Comment by tomasr8** [comment]
> maybe this is something to already include in this PR as well?

Sure, I could include it in this PR.

I'll play around with the qr code generation to see if what's best

**Comment by tomasr8** [comment]
> maybe this is something to already include in this PR as well?

Actually, I think it'll be easier (easier for me :D) if we merge this first and I add the compact QR code format in a followup PR.
The app will have to support both QR code versions in any case, but if we do it separately I don't have to maintain two separate versions of looking up registrations (old one using `registration_id` & new one using `ticket_uuid`)

File: indico/modules/events/registration/controllers/api/checkin.py [hunk 2]
```diff
@@ -107,3 +107,18 @@ def _process_PATCH(self, checked_in=None, paid=None):
...
+    """Get full details for a specific registration from the ticket UUID (checkin secret)."""
+
+    def _process_args(self):
+        ticket_uuid = str(request.view_args['ticket_uuid'])
+        self.registration = (Registration.query
+                             .filter_by(ticket_uuid=ticket_uuid, is_deleted=False)
+                             .first_or_404())
+        self.regform = self.registration.registration_form
+        self.event = self.registration.event
```
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/api/checkin.py:121]:
```python
if self.regform.is_deleted or self.event.is_deleted:
    raise NotFound
```

I don't think this is checked anywhere else...

Comment (ThiefMaster) [reply indico/modules/events/registration/controllers/api/checkin.py:121]: nevermind, event deleted is checked in `RHEventBase`. but for the regform there is no such check anywhere else

**Review by ThiefMaster — APPROVED** [review]

**Pull request merged**